metamusic/clients/songmetadata.py

```diff
@@ -0,0 +1,123 @@
+#!/usr/bin/env python3
+"""
+Author : Salvador <Salvador@fedora>
+Date   : 2022-10-15
+Purpose: New program
+"""
+from metamusic.models import Track
+import argparse
+import eyed3
+
+
+# --------------------------------------------------
+def get_args():
+    """Get command-line arguments"""
+
+    parser = argparse.ArgumentParser(
+        description="eyeD3 client to get or update metadata",
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter,
+    )
+
+    parser.add_argument(
+        "-t",
+        "--track",
+        help="A track file to get or update metadata",
+        metavar="str",
+        type=str,
+        required=True,
+    )
+
+    parser.add_argument(
+        "-i",
+        "--image",
+        help="An image file to update album artwork",
+        metavar="str",
+        type=str,
+        required=True,
+    )
+
+    return parser.parse_args()
+
+
+# --------------------------------------------------
+def get_track_tags(path: str):
+    track = eyed3.load(path)
+    track_info = Track(
+        artist=track.tag.artist,
+        album=track.tag.album,
+        title=track.tag.title,
+        track_number=track.tag.track_num[0],
+        total_tracks=track.tag.track_num[1],
+        album_type=track.tag.album_type,
+        release_date=track.tag.release_date.year,
+    )
+    return track_info
+
+
+# --------------------------------------------------
+def set_selected_tags(path: str, metadata: dict, tag_selection: list[str]):
```

Comment:
thought: You are modifying a variable passed as argument in and outside the function without returning it.

This is purely your choice on how you want things to work, but be aware that you are modifying the metadata argument inside the function and using it outside as it was modified. This can some times get a bit confusing as regularly one expects to get a new object returned by the function. 

However you are returning a None and modifying in place the object you are interested in. This is fine, OpenCV does this for example, and I assume it does it this way for avoiding copying the object that stores image info multiple times.

Just be careful. Write some documentation (docstrings) so you remember the behavior of your function without having to dig into the code to find out.

You can read a bit more of this here
https://stackoverflow.com/questions/575196/why-can-a-function-modify-some-arguments-as-perceived-by-the-caller-but-not-oth

Reply:
Ah, nvm. I thought you were using metadata outside with some modifications done in here. Still, the problem is a little interesting and some times it´s good to know about it :)